### ISSUE TITLE
TypeError: Class extends value undefined is not a constructor or null

### ISSUE THREAD

**Issue by Gesugao-san** [TooTallNate/node-socks-proxy-agent#76]
Just a week ago I installed Windows 21H2 (build 19044.1415). And the other day I needed to install Node.js (16.13.1, npm 8.1.2). After installation, npm refuses to start:

<details>

```
C:\Program Files\nodejs>npm
TypeError: Class extends value undefined is not a constructor or null
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\socks-proxy-agent\dist\agent.js:114:44)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\socks-proxy-agent\dist\index.js:5:33)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\make-fetch-happen\lib\agent.js:161:25)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
TypeError: Class extends value undefined is not a constructor or null
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\socks-proxy-agent\dist\agent.js:114:44)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\socks-proxy-agent\dist\index.js:5:33)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\make-fetch-happen\lib\agent.js:161:25)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)

C:\Program Files\nodejs>
```

</details>

Original problematic folder here:

<details>

[socks-proxy-agent.zip](https://github.com/TooTallNate/node-socks-proxy-agent/files/7801332/socks-proxy-agent.zip)

</details>

In hearts, I deleted the problematic file: https://stackoverflow.com/a/70552439/8175291

<details>

```
CD /D "C:\Program Files\nodejs\node_modules\npm\node_modules"
RMDIR /S /Q "%CD%\socks-proxy-agent\dist\"
npm --global --force install socks-proxy-agent@6.1.1
node --trace-deprecation socks-proxy-agent
```

</details>

Although the npm worked, however, it became impossible to install any other module ("node install --save --force libxmljs socks-proxy-agent osmosis", i was trying [this](https://hackprogramming.com/web-scraping-in-node-js-with-multiple-examples/)). I suspect it is because of these actions.

[2022-01-02T16_09_26_173Z-debug-0.log](https://github.com/TooTallNate/node-socks-proxy-agent/files/7799426/2022-01-02T16_09_26_173Z-debug-0.log)

---

Please help, I'm stumped.

**Comment by zolk** [comment]
I just ran into this same problem in a CircleCI container. The container image I was using shipped with NodeJS 14.17.2. Upgrading to NodeJS 16.13.1 would break npm with the same error you’re receiving. I was able to solve this by deleting the global node_modules folder prior to the Node 16 upgrade (`/usr/local/lib/node_modules`, in my case; I’m not sure off the top of my head where this folder is located in Windows). This fixed the problem.

An alternative solution seems to be to install NPM 8 _before_ upgrading NodeJS, but deleting the global node_modules folder is a simpler route that should do the trick.

**Comment by TooTallNate** [comment]
This code in this repository has been moved to the [`proxy-agents` monorepo](https://github.com/TooTallNate/proxy-agents), so I am closing this pull request. If you feel that this issue still exists as of the latest release, feel free to open a new issue over there.
